doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,143 @@
+---
+title: "PARTITION BY"
+description: "Specifying the sort order of the data."
+aliases:
+  - /guides/partition-by/
+  - /sql/patterns/partition-by/
+menu:
+  main:
+    parent: 'sql-patterns'
+---
+
+A few types of Materialize collections are durably written to storage: [materialized views](/sql/create-materialized-view/), [tables](/sql/create-table), and [sources](/sql/create-source).
+
+Internally, Materialize stores these durable collections in an [LSM-tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree)-like structure. Each collection is made up of a set of
+**runs** of data, each run is sorted and then split up into individual **parts**, and those parts are written to object storage and retrieved only when necessary to satisfy a query. Materialize will also periodically **compact** the data it stores to consolidate small parts into larger ones or discard deleted rows.
+
+Materialize lets you specify the ordering it will use to sort these runs of data internally. A well-chosen sort order can unlock optimizations like [filter pushdown](#filter-pushdown), which in turn can make queries and other operations more efficient.
+
+## Syntax
+
+The option `PARTITION BY <column list>` declares that a [materialized view](/sql/create-materialized-view/#with_options), [table](/sql/create-table/#with_options), or source table should be ordered by the listed columns. For example, a table that stores an append-only collection of events may look like:
+
+```mzsql
+CREATE TABLE events (created_at timestamptz, body jsonb)
+WITH (
+    PARTITION BY (created_at)
+);
+```
+
+When multiple columns are specified, rows are ordered lexicographically:
+`PARTITION BY (created_date, created_time)` would sort first by the created date, then order rows with equal created date by time. Durable collections without a `PARTITION BY` option might be stored in any order.
+
+Note that this declaration has no impact on the order in which records are returned by queries. If you want to return results in a specific order, use an
```

Comment:
Would just use a note box to draw more attention to this

```suggestion
{{< note >}}
The `PARTITION BY` option has no impact on the order in which records are returned by queries. If you want to return results in a specific order, use an `ORDER BY` clause on your select statement.
{{< /note >}}
```

We might also want to repeat the note up top because people will skip over it.